src/main/java/org/truffleruby/core/queue/QueueNodes.java

```diff
@@ -73,18 +70,11 @@ protected RubyQueue push(RubyQueue self, final Object value) {
 
     }
 
-    @CoreMethod(names = { "pop", "shift", "deq" }, optional = 1)
-    @NodeChild(value = "queue", type = RubyNode.class)
-    @NodeChild(value = "nonBlocking", type = RubyBaseNodeWithExecute.class)
-    public abstract static class PopNode extends CoreMethodNode {
-
-        @CreateCast("nonBlocking")
-        protected RubyBaseNodeWithExecute coerceToBoolean(RubyBaseNodeWithExecute nonBlocking) {
-            return BooleanCastWithDefaultNode.create(false, nonBlocking);
-        }
```

Comment:
Does it work without such explicit casting of `nonBlocking` parameter to `boolean` when actual argument value isn't Ruby `true`/`false`?

Reply:
True, I'll fix that with `Primitive.as_boolean`

Reply:
Again, if this case isn't covered with specs - it's a good moment to add a new test.

Reply:
On it.